### PR TITLE
UOE-11168: Consume then Remove `app.id` from ApplovinMax Request

### PR DESCRIPTION
**Pull request by pm-saurabh-narkhede**

### CODE REVIEW

**Review by AvinashKapre — APPROVED** [review]
LGTM

File: modules/pubmatic/openwrap/applovinmax.go [hunk 1]
```diff
@@ -251,6 +252,23 @@ func updateAppLovinMaxRequest(requestBody []byte, rctx models.RequestCtx) []byte
 	return requestBody
 }
 
+func setProfileID(requestBody []byte) ([]byte, string) {
+	if profileId, err := jsonparser.GetInt(requestBody, "ext", "prebid", "bidderparams", "pubmatic", "wrapper", "profileid"); err == nil && profileId != 0 {
+		return requestBody, strconv.Itoa(int(profileId))
+	}
+
+	profIDStr, err := jsonparser.GetString(requestBody, "app", "id")
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/applovinmax.go:260]: We are not handling the case where profile id is passed with some string value like app.id = "abc", then that value will be set as profile ID. Please handle that case.

File: modules/pubmatic/openwrap/applovinmax.go [hunk 2]
```diff
@@ -251,6 +252,23 @@ func updateAppLovinMaxRequest(requestBody []byte, rctx models.RequestCtx) []byte
...
+	if err != nil || len(profIDStr) == 0 {
+		return requestBody, ""
+	}
+
+	requestBody = jsonparser.Delete(requestBody, "app", "id")
+	if newRequestBody, err := jsonparser.Set(requestBody, []byte(profIDStr), "ext", "prebid", "bidderparams", "pubmatic", "wrapper", "profileid"); err == nil {
```
[the remark this answers is not in the document]
Comment (pm-saurabh-narkhede) [reply modules/pubmatic/openwrap/applovinmax.go:272]: Done

File: modules/pubmatic/openwrap/applovinmax.go (hunk 1 above)
Comment (pm-saurabh-narkhede) [reply modules/pubmatic/openwrap/applovinmax.go:261]: Done

**Review by AvinashKapre — DISMISSED** [review]

**Review by pm-isha-bharti — APPROVED** [review]

**Review by AvinashKapre — APPROVED** [review]

**Pull request merged**